base/system/winlogon/sas.c

```diff
@@ -571,6 +578,7 @@ HandleLogon(
     /* Logon has succeeded. Play sound. */
     PlayLogonSound(Session);
 
+    SetLogonTimestamp(Session);
```

Comment:
Maybe you should add a comment that the timestamp has to be set after the PlayLogonSound for being able to distinguish between first-logon and others.

Reply:
OK.